src/services/pairing/usePairing.ts

```diff
@@ -0,0 +1,112 @@
+import { useState, useEffect, useCallback } from 'react'
+
+import useChainId from '@/hooks/useChainId'
+import useOnboard, { getConnectedWallet } from '@/hooks/wallets/useOnboard'
+import { logError, Errors } from '@/services/exceptions'
+import { getPairingConnector, WalletConnectEvents } from '@/services/pairing/connector'
+import { PAIRING_MODULE_LABEL } from '@/services/pairing/module'
+import { formatPairingUri, killPairingSession } from '@/services/pairing/utils'
+
+const connector = getPairingConnector()
+
+/**
+ * `useInitPairing` is responsible for WC session management, creating a session when:
+ *
+ * - no wallet is connected to onboard, deemed "intializing" pairing (disconnecting wallets via the UI)
+ * - on WC 'disconnect' event (disconnecting via the app)
+ */
+
+// First session will be created by onboard's state subscription, only
+// when there is no connected wallet
+let hasInitialized = false
+
+// WC has no flag to determine if a session is currently being created
+let isConnecting = false
+
+export const useInitPairing = () => {
+  const onboard = useOnboard()
+  const chainId = useChainId()
+
+  const canConnect = !connector.connected && !isConnecting
+
+  const createSession = useCallback(() => {
+    if (!canConnect) {
+      return
+    }
+
+    isConnecting = true
+    connector.createSession({ chainId: +chainId }).then(() => {
+      isConnecting = false
+    })
+  }, [canConnect, chainId])
+
+  useEffect(() => {
+    if (!onboard) {
+      return
+    }
+
+    // Upon successfuly WC connection, connect it to onboard
+    connector.on(WalletConnectEvents.CONNECT, () => {
+      onboard
+        .connectWallet({
+          autoSelect: {
+            label: PAIRING_MODULE_LABEL,
+            disableModals: true,
+          },
+        })
+        .catch((e) => logError(Errors._302, (e as Error).message))
+    })
+
+    // Create new session when app disconnects
```

Comment:
This comment seems redundant as it just describes what the next two lines do which is already readable imo.

Reply:
I've adjusted it to "Create new session when the mobile app disconnects" to differentiate it from UI-related disconnection. Do you think it should be removed entirely?

Reply:
As per our discussion, we'll remove this.